wasm/v8/module.go

```diff
@@ -97,54 +93,81 @@ func runJS(inst *V8Instance, code, filename string) error {
 	return nil
 }
 
-func callHandlers(inst *V8Instance, call *wasm.Call) error {
+// callHandlers determines the type of handler (map, store) for a given module name and dispatches the execution to the appropriate handler function.
+func callHandlers(inst *V8Instance, call *wasm.Call, input []byte) error {
 	handlerName := call.ModuleName
 
-	// Check what type of handler this is
-	checkTypeCode := fmt.Sprintf(`getHandlerType("%s")`, handlerName)
-	handlerTypeVal, err := inst.ctx.RunScript(checkTypeCode, "check_handler_type.js")
+	// Check the type of handler (map, store)
+	typeCheckVal, err := inst.ctx.Global().Get("getHandlerType")
 	if err != nil {
-		return fmt.Errorf("failed to check handler type: %w", err)
+		return fmt.Errorf("missing getHandlerType: %w", err)
 	}
-
-	if handlerTypeVal.IsNull() || handlerTypeVal.IsUndefined() {
-		return fmt.Errorf("handler '%s' not found", handlerName)
+	getHandlerType, err := typeCheckVal.AsFunction()
+	if err != nil {
+		return fmt.Errorf("getHandlerType not a function: %w", err)
+	}
+	nameVal, _ := v8go.NewValue(inst.ctx.Isolate(), handlerName)
+	handlerTypeVal, err := getHandlerType.Call(v8go.Undefined(inst.ctx.Isolate()), nameVal)
+	if err != nil {
+		return fmt.Errorf("getHandlerType call failed: %w", err)
 	}
 
-	handlerType := handlerTypeVal.String()
-
-	switch handlerType {
+	// Dispatch to the appropriate handler
+	switch handlerTypeVal.String() {
 	case "map":
-		return callMapHandler(inst, handlerName)
+		return callMapHandler(inst, handlerName, input)
 	case "store":
-		return callStoreHandler(inst, call, handlerName)
+		return callStoreHandler(inst, handlerName, input)
 	default:
-		return fmt.Errorf("unknown handler type: %s", handlerType)
+		return fmt.Errorf("unknown handler type: %s", handlerTypeVal.String())
```

Comment:
```suggestion
		return fmt.Errorf("unknown handler type: %s", handlerTypeVal)
```
`%s` calls `.String()` automatically if defined.